zarr/storage.py

```diff
@@ -176,7 +176,7 @@ def getsize(store, path: Path = None) -> int:
     if hasattr(store, 'getsize'):
         # pass through
         return store.getsize(path)
-    elif isinstance(store, dict):
```

Comment:
I don't see why we should limit ourselves to dict here, it shoudl work with any mutable mappings.

Reply:
Yeah this seems fine.